livekit-plugins/livekit-plugins-deepgram/livekit/plugins/deepgram/stt.py

```diff
@@ -776,3 +777,36 @@ def _validate_model(
         )
         return "nova-2-general"
     return model
+
+
+def _validate_keyterms(
+    model: DeepgramModels | str,
+    language: NotGivenOr[DeepgramLanguages | str],
+    keyterms: NotGivenOr[list[str]],
+    keywords: NotGivenOr[list[tuple[str, float]]],
+) -> tuple[
+    NotGivenOr[list[str]],
+    NotGivenOr[list[tuple[str, float]]],
+]:
+    """
+    Validating keyterms and keywords for model compatibility.
+    See: https://developers.deepgram.com/docs/keyterm and https://developers.deepgram.com/docs/keywords
+    """
+    if model.startswith("nova-3") and is_given(keywords):
+        logger.warning(
```

Comment:
should it throw an error instead?

this seems like a critical error for those that rely on this behavior.

Reply:
That makes sense. I will change it.